src/components/home/sprint-view-card-content.tsx/user-sprint-view-card.tsx

```diff
@@ -0,0 +1,116 @@
+import { useAtom, useAtomValue } from "jotai";
+import { useEffect, useState } from "react";
+import { personsAtom } from "../../../atoms/person";
+import { userProfileAtom } from "../../../atoms/auth";
+import { Person } from "../../../generated/client";
+import config from "../../../app/config";
+import { useLambdasApi } from "../../../hooks/use-api";
+import { Allocations, Projects, TimeEntries } from "../../../generated/homeLambdasClient";
+import { CardContent, CircularProgress, Typography } from "@mui/material";
+import SprintViewBarChart from "../../charts/sprint-view-bar-chart";
+import { SprintViewChartData } from "../../../types";
+import strings from "../../../localization/strings";
+
+/**
+ * Sprint card component for users
+ */
+const UserSprintViewCard = () => {
+  const [loading, setLoading] = useState(false);
+  const [persons] = useAtom(personsAtom);
+  const userProfile = useAtomValue(userProfileAtom);
+  const loggedInPerson = persons.find(
+    (person: Person) => person.id === config.person.forecastUserIdOverride || person.keycloakId === userProfile?.id
+  );
+  const [allocations, setAllocations] = useState<Allocations[]>([]);
+	const [projects, setProjects] = useState<(Projects|undefined)[]>([]);
+	const [timeEntries, setTimeEntries] = useState<number[]>([]);
+  const {allocationsApi, projectsApi, timeEntriesApi} = useLambdasApi();
+
+	useEffect(()=>{
+		getAllocationsAndProjects();
+	},[loggedInPerson]);
+
+	/**
+   * Get allocations, projects and time entries
+   */
+	const getAllocationsAndProjects = async () => {
+		setLoading(true);
+		if (loggedInPerson) {
+			try {
+				const fetchedAllocations = await allocationsApi.listAllocations({personId: loggedInPerson.id.toString(), startDate: new Date()});
+				const fetchedProjects = await projectsApi.listProjects();
+				const findProjects =fetchedAllocations.map( (allocation)=>{
+					return fetchedProjects.find((project) => project.id === allocation.project) || undefined;
+				});
+
+				const totalTimeEntries = await Promise.all(fetchedAllocations.map(async(allocation) => {
+					try {
+						const fetchedTimeEntries = await timeEntriesApi.listProjectTimeEntries({ projectId: allocation.project || 0, startDate: allocation.startDate, endDate: allocation.endDate});
+						let totalMinutes = 0;
+            fetchedTimeEntries.forEach((timeEntry: TimeEntries) => {
+              if (loggedInPerson && timeEntry.person === loggedInPerson.id) {
+                totalMinutes += (timeEntry.timeRegistered || 0)
+              }
+            }); 
+            return totalMinutes;
+					} catch (error){
+						console.error("Error fetching time entries:", error);
+						
+					}
+					return 0;
+				}));
+				setAllocations(fetchedAllocations);
+				setProjects(findProjects);
```

Comment:
How about filtering "findProjects" so that it removes any undefined values. This would allow you to type the useState as useState<Projects[]>([]); instead of useState<(Projects|undefined)[]>([]);